spec/lib/miq_expression/query_helper_spec.rb

```diff
@@ -0,0 +1,142 @@
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id))
+      expect(result).to eq(expected)
+    end
+
+    # query:  "hosts"."type" = 'abc' AND "vms"."host_id" = "hosts"."id" AND "hosts"."type" = 'def'
+    # result: "hosts"."type" = 'abc' AND "hosts"."type" = 'def'
+    it "removes matching query in the middle of a tripple AND" do
+      query = arel_and(host_type.eq('abc'), vm_host_id.eq(host_id), host_type.eq('def'))
+      expected = host_type.eq('abc').and(host_type.eq('def'))
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id), [])
+      expect(result).to eq(expected)
+    end
+
+    # query:  "vms"."host_id" = "hosts"."id" AND "hosts"."type" = 'abc' AND "hosts"."type" = 'def'
+    # result: "hosts"."type" = 'abc' AND "hosts"."type" = 'def'
+    it "removes matching query at the end of a tripple AND" do
+      query = vm_host_id.eq(host_id).and(host_type.eq('abc').and(host_type.eq('def')))
+      expected = host_type.eq('abc').and(host_type.eq('def'))
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id))
+      expect(result).to eq(expected)
+    end
+
+    # query:  "hosts"."type" = 'abc' AND "vms"."host_id" = "hosts"."id" AND "hosts"."type" = 'def'
+    # result: "hosts"."type" = 'abc' AND "hosts"."type" = 'abc'
+    it "removes matching query from multiple layers where our equality is at the base" do
+      query = host_type.eq('abc').and(vm_host_id.eq(host_id).and(host_type.eq('def')))
+      expected = host_type.eq('abc').and(host_type.eq('def'))
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id))
+      expect(result).to eq(expected)
+    end
+
+    # query:  ("hosts"."type" = 'abc') AND "vms"."host_id" = "hosts"."id"
+    # result: ("hosts"."type" = 'abc')
+    it "removes matching query with grouped other node" do
+      query = arel_group(host_type.eq('abc')).and(vm_host_id.eq(host_id))
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id), [])
+      expect(result).to eq(arel_group(host_type.eq('abc')))
+    end
+
+    # query:  "hosts"."type" = 'abc' AND ("vms"."host_id" = "hosts"."id")
+    # result: "hosts"."type" = 'abc'
+    it "removes matching node with grouped matchng node" do
+      query = host_type.eq('abc').and(arel_group(vm_host_id.eq(host_id)))
+      result = described_class.remove_equality(query, vm_host_id.eq(host_id))
+      expect(result).to eq(host_type.eq('abc'))
+    end
+
+    private
+
+    def bind_values(*values)
+      values.map do |value|
+        type = value.kind_of?(String) ? ActiveModel::Type::String.new : ActiveModel::Type::Integer.new
+        if ActiveRecord.version.to_s > "5.2"
+          ActiveModel::Attribute.from_database('param', value, type)
+        else
+          ActiveRecord::Attribute.from_database('param', value, type)
+        end
+      end
+    end
+
+    # this helps us greate some of the odd cases we've seen around
```

Comment:
"this helps us greate" ??

Is this supposed to be "generate"?  Probably not meant to be "great"...

Reply:
create great case?